test/api_integration/apis/saved_objects/update.ts

```diff
@@ -96,6 +96,54 @@ export default function ({ getService }: FtrProviderContext) {
         expect(resp.body.references).to.eql([]);
       });
 
+      it('handles upsert', async () => {
+        await supertest
+          .put(`/api/saved_objects/visualization/upserted-viz`)
+          .send({
+            attributes: {
+              title: 'foo',
+            },
+            upsert: {
+              description: 'upserted description',
+              version: 42,
+            },
+          })
+          .expect(200);
```

Comment:
can we add a test for when the keys in `attributes` and `upsert` collide?

Reply:
as soon as we make a decision about https://github.com/elastic/kibana/pull/98712#discussion_r623072556!